viewing_party/party.py

```diff
@@ -1,23 +1,151 @@
+        all_genres.append(already_watched[i]["genre"])
+        top_genre = [word for word, word_count in Counter(all_genres).most_common(1)]
+        top_genre = top_genre[0]
+        return top_genre
+     if already_watched == []:
+        top_genre = None
+     return top_genre
+     
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    
+    friends_not_watched = []
+    friends_watched = []
+    user_watched = user_data["watched"]
+
+    for movie in user_watched:
+        friends_not_watched.append(movie)
+
+    for friend_list in user_data["friends"]:
+        for friend in friend_list["watched"]:
+            friends_watched.append(friend)
+    for i in friends_watched:
+        if i in friends_not_watched:
+            friends_not_watched.remove(i)
+    return friends_not_watched
+
+def get_friends_unique_watched(user_data):
+   
+    only_friends_watched = []
+    already_watched = []
+    user_watched = user_data["watched"]
+
+    for friend_list in user_data["friends"]:
+        for friend in friend_list["watched"]:
+            if friend not in only_friends_watched:
+                only_friends_watched.append(friend)
+   
+    for movie in user_watched:
+        already_watched.append(movie)
+    
+    for i in already_watched:
+        if i in only_friends_watched:
+            only_friends_watched.remove(i)
+    
+    return only_friends_watched
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    
+    recommendations = []
+    user_subscriptions = user_data["subscriptions"]
+    friends_watch_list = get_friends_unique_watched(user_data)
+    
+    for m in range(len(friends_watch_list)):
+        if friends_watch_list[m]["host"] in user_subscriptions:
+            recommendations.append(friends_watch_list[m])
 
+    return recommendations
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
-# -----------------------------------------
+# ----------------------------------------
+def get_new_rec_by_genre(user_data):
+    
+    recommendations_genre = []
+    favorite_genre = get_most_watched_genre(user_data)
+    friends_watch_list = get_friends_unique_watched(user_data)
+
+    for m in range(len(friends_watch_list)):
+        if friends_watch_list[m]["genre"] ==  favorite_genre:
+            recommendations_genre.append(friends_watch_list[m])
+    return recommendations_genre
+
+def get_rec_from_favorites(user_data):
+
+    recommendations_favs = []
+    favorite_movies = user_data["favorites"]
+    user_watch_list = get_unique_watched(user_data)
 
+    for movie in user_watch_list:
+        if movie in favorite_movies:
+            recommendations_favs.append(movie)
+    return recommendations_favs

```

Comment:
👍  Yess! Helper functions!! Just like the previous function, how would this function look if we used list comprehension?